src/map/mob.c

```diff
@@ -5367,70 +5363,109 @@ static void mob_mobavail_removal_notice(void)
-			for(j=0,p=line;j<3 && p;j++){
-				str[j]=p;
-				p=strchr(p,',');
-				if(p) *p++=0;
-			}
+	const char *filename[] = {
+		DBPATH"mob_group.conf",
+		"mob_group2.conf"
+	};
 
-			if(str[0]==NULL || str[2]==NULL)
-				continue;
+	for (int i = 0; i < ARRAYLENGTH(filename); i++)
+		mob->read_group_db_libconfig(filename[i]);
+}
 
-			class_ = atoi(str[0]);
-			if(mob->db(class_) == mob->dummy)
-				continue;
-			count++;
-			mob->db_data[class_]->summonper[i]=atoi(str[2]);
-			if (i) {
-				if( summon[i].qty < ARRAYLENGTH(summon[i].class_) ) //MvPs
-					summon[i].class_[summon[i].qty++] = class_;
-				else {
-					ShowDebug("Can't store more random mobs from %s, increase size of mob.c:summon variable!\n", mobfile[i]);
-					break;
-				}
-			}
+static bool mob_read_group_db_libconfig(const char *filename)
+{
+	struct config_t mg_conf;
+	char filepath[256];
+
+	safesnprintf(filepath, sizeof(filepath), "%s/%s", map->db_path, filename);
+	if (libconfig->load_file(&mg_conf, filepath) == CONFIG_FALSE) {
+		ShowError("%s: can't read %s\n", __func__, filepath);
+		return false;
+	}
+
+	int i = 0;
+	int count = 0;
+	struct config_setting_t *it = NULL;
+
+	while ((it = libconfig->setting_get_elem(mg_conf.root, i++)) != NULL) {
+		if (mob->read_group_db_libconfig_sub(it, filepath))
+			++count;
+	}
+
+	libconfig->destroy(&mg_conf);
+	ShowStatus("Done reading '"CL_WHITE"%d"CL_RESET"' entries in '"CL_WHITE"%s"CL_RESET"'.\n", count, filepath);
+	return true;
+}
+
+static bool mob_read_group_db_libconfig_sub(struct config_setting_t *it, const char *source)
+{
+	nullpo_retr(false, it);
+	nullpo_retr(false, source);
+
+	const char *name = config_setting_name(it);
+	int group_id;
+
+	if (!script->get_constant(name, &group_id)) {
+		ShowWarning("%s: unknown monster group '%s' in \"%s\", skipping..\n", __func__, name, source);
+		return false;
+	}
+
+	if (!config_setting_is_list(it)) {
+		ShowWarning("%s: monster group '%s' in \"%s\" must be a list, skipping..\n", __func__, name, source);
+		return false;
+	}
+
+	if (!mob->read_group_db_libconfig_sub_group(it, group_id, source))
+		return false;
+
+	return true;
+}
+
+static bool mob_read_group_db_libconfig_sub_group(struct config_setting_t *it, enum mob_groups group_id, const char *source)
+{
+	nullpo_retr(false, it);
+	nullpo_retr(false, source);
+	Assert_retr(false, group_id >= 0 && group_id < MOBG_MAX_GROUP);
+
+	if (mob->mob_groups[group_id] == NULL) {
+		mob->mob_groups[group_id] = aCalloc(1, sizeof(struct mob_group));
+	} else {
+		VECTOR_CLEAR(*mob->mob_groups[group_id]);
+	}
+	VECTOR_INIT(*mob->mob_groups[group_id]);
+
+	int i = 0;
+	struct config_setting_t *entry = NULL;
+
+	while ((entry = libconfig->setting_get_elem(it, i++)) != NULL) {
+		int class_ = 0;
+		const char *name = libconfig->setting_get_string_elem(entry, 0);
+
+		if (!script->get_constant(name, &class_)) {
+			ShowWarning("%s: Invalid monster '%s' in \"%s\", skipping..\n", __func__, name, source);
+			return false;
```

Comment:
This should probably `continue` instead of returning since you wrote "skipping" in the message above